chips/nrf52/src/qdec.rs

```diff
@@ -0,0 +1,287 @@
+//!  Qdec driver, nRF5x-family
+//!  set_client(), enable, get_ticks,
+//!  The nRF5x quadrature decoder
+//!
+#[allow(unused_imports)]
+use core;
+use kernel::common::cells::OptionalCell;
+use kernel::common::registers::{
+    register_bitfields, register_structs, ReadOnly, ReadWrite, WriteOnly,
+};
+use kernel::common::StaticRef;
+use kernel::ReturnCode;
+use nrf5x::pinmux;
+// In this section I declare a struct called QdecRegisters, which contains all the
+// relevant registers as outlined in the Nordic 5x specification of the Qdec.
+register_structs! {
+    pub QdecRegisters {
+        /// Start Qdec sampling
+        (0x000 => tasks_start: WriteOnly<u32, Task::Register>),
+        /// Stop Qdec sampling
+        (0x004 => tasks_stop: WriteOnly<u32, Task::Register>),
+        /// Read and clear ACC and ACCDBL
+        (0x008 => tasks_readclracc: WriteOnly<u32, Task::Register>),
+        /// Read and clear ACC
+        (0x00C => tasks_rdclracc: WriteOnly<u32, Task::Register>),
+        /// Read nad clear ACCDBL
+        (0x010 => tasks_rdclrdbl: WriteOnly<u32, Task::Register>),
+        ///Reserve space so tasks_rdclrdbl has access to its entire address space (?)
+        (0x0014 => _reserved),
+        /// All the events which have interrupts!
+        (0x100 => events_arr: [ReadWrite<u32, Event::Register>; 5]),
+        /// Event being generated for every new sample
```

Comment:
should I remove all these comments here?

Reply:
why are they commented out? It seems like they ought to just be included.

Reply:
because they are already being accounted for as part of the events_arr :)

Reply:
oh, right, in that case yes delete them